### PR TITLE
376989: Field Config rule

### PR DESCRIPTION
**Pull request by kamilmkainos**

### CODE REVIEW

File: src/STW.ProcessingApi.Function/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -52,6 +52,13 @@ public static void RegisterHttpClients(this IServiceCollection serviceCollection
                 client.BaseAddress = new Uri(options.CommodityCodeBaseUrl);
                 client.Timeout = TimeSpan.FromSeconds(options.Timeout);
             });
+        serviceCollection.AddHttpClient<IFieldConfigService, FieldConfigService>(
```
Comment (sean-treanor) [inline src/STW.ProcessingApi.Function/Extensions/ServiceCollectionExtensions.cs:74]: Add a line break

**Review by reece-bennett — APPROVED** [review]

File: src/STW.ProcessingApi.Function/Helpers/FieldConfigHelper.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using Newtonsoft.Json.Linq;
+
+public static class FieldConfigHelper
+{
+    public static List<string> GetValidPurposes(string fieldConfigData, string? speciesTypeName)
+    {
+        var internalMarketKey = string.IsNullOrEmpty(speciesTypeName) ? "internalMarket" : $"internalMarket-{speciesTypeName}";
+        var expression = $"$..['Purpose']..[?(@.id=='{internalMarketKey}' || @.name=='{internalMarketKey}')]..values";
+        var array = (JArray)JToken.Parse(fieldConfigData).SelectTokens(expression).FirstOrDefault() ?? [];
+        var internalMarketPurposes = new List<string>();
```
Comment (sean-treanor) [inline src/STW.ProcessingApi.Function/Helpers/FieldConfigHelper.cs]: Just a suggestion on how you can drop the if statement as the `Select` is the same as `map` in Java.

```csharp
    public static List<string> GetValidPurposes(string fieldConfigData, string? speciesTypeName)
    {
        var internalMarketKey = string.IsNullOrEmpty(speciesTypeName)
            ? "internalMarket"
            : $"internalMarket-{speciesTypeName}";

        var expression = $"$..['Purpose']..[?(@.id=='{internalMarketKey}' || @.name=='{internalMarketKey}')]..values";

        var internalMarketPurposes = JToken.Parse(fieldConfigData)
            .SelectTokens(expression)
            .FirstOrDefault()?
            .Select(x => ((JObject)x).GetValue("label")!.ToString())
            .ToList();

        return internalMarketPurposes ?? [];
    }
```

File: src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    new ValidationError(
+                        string.Format(RuleErrorMessage.FieldConfigServiceError, result.Error.Message),
+                        RuleErrorId.FieldConfigServiceError));
+                return;
+        }
+
+        var speciesTypeName = ComplementHelper.GetSpeciesTypeName(SpsCertificateHelper.GetApplicableSpsClassifications(spsCertificate));
+        var goodsCertifiedAs = PurposeHelper.GetGoodsCertifiedAs(spsCertificate.SpsExchangedDocument.SignatorySpsAuthentication)!;
+        var validPurposes = FieldConfigHelper.GetValidPurposes(result.Value!.Data, speciesTypeName);
+        if (!validPurposes.Contains(goodsCertifiedAs))
```
Comment (sean-treanor) [inline src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs:43]: Add line break

File: src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        var chedType = SpsCertificateHelper.GetChedType(spsCertificate.SpsExchangedDocument.IncludedSpsNote);
+        var purpose = PurposeHelper.GetPurpose(spsCertificate.SpsExchangedDocument.SignatorySpsAuthentication);
+
+        return chedType == ChedType.Chedp && purpose == Purpose.InternalMarket;
+    }
+
+    public async Task InvokeAsync(SpsCertificate spsCertificate, IList<ValidationError> errors)
+    {
+        var result = await FetchFieldConfig(spsCertificate);
+        if (!result.IsSuccess)
```
Comment (sean-treanor) [inline src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs:30]: add line break

File: src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                    RuleErrorId.InvalidInternalMarketPurposeError));
+        }
+    }
+
+    private async Task<Result<FieldConfigDto>> FetchFieldConfig(SpsCertificate spsCertificate)
+    {
+        var chedType = SpsCertificateHelper.GetChedType(spsCertificate.SpsExchangedDocument.IncludedSpsNote)!;
+        var applicableSpsClassification = SpsCertificateHelper.GetApplicableSpsClassifications(spsCertificate);
+        var commodityCode = CommodityHelper.GetCommodityId(applicableSpsClassification)!;
+        return await _fieldConfigService.GetByCertTypeAndCommodityCode(chedType, commodityCode);
```
Comment (sean-treanor) [inline src/STW.ProcessingApi.Function/Rules/Chedp/InternalMarketFieldConfigRule.cs:58]: add line break

**Review by sean-treanor — APPROVED** [review]

**Pull request merged**